generator/src/test/java/com/scottlogic/deg/generator/restrictions/StringRestrictionsTests.java

```diff
@@ -0,0 +1,642 @@
+                .intersect(ofLength(5, false));
+
+        StringGenerator generator = restrictions.createGenerator();
+
+        assertGeneratorCannotGenerateAnyStrings(generator);
+    }
+
+    @Test
+    void createGenerator_withNegatedMaxAndContradictingOfLengthConstraint_shouldCreateStringsShorterThanMaximumLength() {
+        StringRestrictions restrictions =
+            maxLength(4)
+                .intersect(ofLength(10, true));
+
+        StringGenerator generator = restrictions.createGenerator();
+
+        Assert.assertThat(generator.toString(), equalTo("/^.{0,4}$/"));
+    }
+
+    @Test
+    void createGenerator_withNegatedMinMaxAndNonContradictingOfLengthConstraint_should() {
+        StringRestrictions restrictions =
+            maxLength(5)
+                .intersect(minLength(10))
+                .intersect(ofLength(7, true));
+
+        StringGenerator generator = restrictions.createGenerator();
+
+        assertGeneratorCannotGenerateAnyStrings(generator);
+    }
+
+    @Test
+    void createGenerator_withNegated2MinLengthConstraints_shouldCreateStringsUptoShortestLength() {
+        StringRestrictions restrictions =
+            maxLength(5)
+                .intersect(maxLength(10));
+
+        StringGenerator generator = restrictions.createGenerator();
+
+        Assert.assertThat(generator.toString(), equalTo("/^.{0,5}$/"));
+    }
+
+    @Test
+    void createGenerator_withNegated2MaxLengthConstraints_shouldCreateStringsFromShortestLengthToDefaultMax() {
+        StringRestrictions restrictions =
+            minLength(5)
+                .intersect(minLength(10));
+
+        StringGenerator generator = restrictions.createGenerator();
+
+        Assert.assertThat(generator.toString(), equalTo("/^.{10,}$/"));
+    }
+
+    @Test
+    void createGenerator_with2OfDifferentLengthConstraintsWhereOneIsNegated_shouldCreateStringsOfNonNegatedLength() {
+        StringRestrictions restrictions =
+            ofLength(5, false)
+                .intersect(ofLength(10, true));
+
+        StringGenerator generator = restrictions.createGenerator();
+
+        Assert.assertThat(generator.toString(), equalTo("/^.{5}$/"));
+    }
+
+    @Test
+    void createGenerator_with2OfLengthConstraintsWhereOneIsNegated_should() {
+        StringRestrictions restrictions =
+            ofLength(5, false)
+                .intersect(ofLength(5, true));
+
+        StringGenerator generator = restrictions.createGenerator();
+
+        assertGeneratorCannotGenerateAnyStrings(generator);
+    }
+
+    @Test
+    void createGenerator_withNotOfLengthSameAsMaxLength_shouldPermitStringsUpToMaxLengthLess1() {
+        StringRestrictions restrictions =
+            maxLength(5)
+                .intersect(ofLength(4, true));
+
+        StringGenerator generator = restrictions.createGenerator();
+
+        Assert.assertThat(generator.toString(), equalTo("/^.{0,3}$/"));
+    }
+
+    private static StringRestrictions ofLength(int length, boolean negate){
```

Comment:
Should these be replaced with calls to the factory?

Reply:
Possibly, but it could detract from the testing of the `StringRestrictions` type in isolation of other types.